### PR TITLE
[BugFix] `torch.where`: Expand mask on the right in lazy stack tds

### PR DESCRIPTION
**Pull request by vmoens**
Masks in `torch.where` for lazy stacks need to have as many dims as the tensordict to be unbound along the stack dim. This PR makes sure that it's the case.

### CODE REVIEW

**Review by matteobettini — APPROVED** [review]
LGTM
